syz-ci/manager.go

```diff
@@ -482,68 +481,30 @@ func (mgr *Manager) restartManager() {
 }
 
 func (mgr *Manager) testImage(imageDir string, info *BuildInfo) error {
+	testSem.Wait()
+	defer testSem.Signal()
+
 	log.Logf(0, "%v: testing image...", mgr.name)
 	mgrcfg, err := mgr.createTestConfig(imageDir, info)
 	if err != nil {
 		return fmt.Errorf("failed to create manager config: %w", err)
 	}
-	if !vm.AllowsOvercommit(mgrcfg.Type) {
-		return nil // No support for creating machines out of thin air.
-	}
-	osutil.MkdirAll(mgrcfg.Workdir)
-	configFile := filepath.Join(mgrcfg.Workdir, "manager.cfg")
-	if err := config.SaveFile(configFile, mgrcfg); err != nil {
-		return err
-	}
-
-	testSem.Wait()
-	defer testSem.Signal()
-
-	timeout := 30 * time.Minute * mgrcfg.Timeouts.Scale
-	bin := filepath.Join(mgrcfg.Syzkaller, "bin", "syz-manager")
-	output, retErr := osutil.RunCmd(timeout, "", bin, "-config", configFile, "-mode=smoke-test")
-	if retErr == nil {
-		return nil
-	}
-
-	var verboseErr *osutil.VerboseError
-	if errors.As(retErr, &verboseErr) {
-		// Caller will log the error, so don't include full output.
-		retErr = errors.New(verboseErr.Title)
+	rep, err := instance.RunSmokeTest(mgrcfg)
```

Comment:
```
if err != nil {
 ...
 return ...
}
code using rep
```
looks more idiomatic and saves 1 tab space for the happy path code

Reply:
In this case, the combination of `err != nil` and `rep != nil` is possible and we'd like to still report the `rep` in this case.

Reply:
Though I guess we could rewrite `RunSmokeTest` to always return either one or another

Reply:
Done so. PTAL.